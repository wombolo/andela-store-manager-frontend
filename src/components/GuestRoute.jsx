import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const GuestRoute = ({component: Component, auth, ...rest}) => (
    <Route
        {...rest}
        render={props => (!auth.isAuthenticated ? (<Component {...props} />) : (
          <Redirect to={{
            pathname: '/dashboard',
            state: { from: props.location },
          }}/>
          ))}
    />
);

GuestRoute.propTypes = {
  auth: PropTypes.shape({}).isRequired,
  location: PropTypes.shape({}).isRequired,
  component: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  auth: state.auth,
});
export default connect(mapStateToProps)(GuestRoute);
