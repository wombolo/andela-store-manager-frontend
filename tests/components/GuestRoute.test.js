import 'babel-polyfill'
import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';

import { GuestRoute } from '../../src/components/GuestRoute.jsx';
import {MemoryRouter} from "react-router-dom";

let wrapper;

const props = {
  component: () => null,
  auth: {
    user: {
      profile: {
        id: 1,
        firstname: 'Akano',
        lastname: 'Peter',
        email: 'elkenzom1@gmail.com',
        role: 'admin',
        image: 'team-7.jpg',
        password: null
      },
      iat: 1551952568,
      exp: 1552038968
    },
    isAuthenticated: true,
    error: '',
    isLoading: false
  },
  location:{}
};


describe('GuestRoute Component', () => {
  it('should match snapshot', () => {
    wrapper = shallow(<GuestRoute {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Methods in Class', () => {
    beforeEach(() => {
      wrapper = mount(<MemoryRouter><GuestRoute {...props}/></MemoryRouter>);
    });

    afterEach(() => {
      wrapper.unmount();
    });


    it('triggers when user is not logged in', () => {
      props.auth.isAuthenticated = false;
      const newWrapper = mount(<MemoryRouter><GuestRoute {...props}/></MemoryRouter>);
    });

  });
});


