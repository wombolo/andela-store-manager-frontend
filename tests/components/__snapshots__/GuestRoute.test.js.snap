// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestRoute Component should match snapshot 1`] = `
<Route
  location={Object {}}
  render={[Function]}
/>
`;
